test/staking/SwapHelper.test.ts

```diff
@@ -0,0 +1,403 @@
+    setDAOAddress("DAI", dai.address);
+
+    //This set addresses should be another function because when we put this initialization of addresses in initializer then nameservice is not ready yet so no proper addresses
+    await goodReserve.setAddresses();
+    const gasFeeMockFactory = await ethers.getContractFactory(
+      "GasPriceMockOracle"
+    );
+    gasFeeOracle = await gasFeeMockFactory.deploy();
+    const daiEthPriceMockFactory = await ethers.getContractFactory(
+      "DaiEthPriceMockOracle"
+    );
+    daiEthOracle = await daiEthPriceMockFactory.deploy();
+
+    const ethUsdOracleFactory = await ethers.getContractFactory(
+      "EthUSDMockOracle"
+    );
+
+    batUsdOracle = await tokenUsdOracleFactory.deploy();
+    ethUsdOracle = await ethUsdOracleFactory.deploy();
+    await dai["mint(address,uint256)"](
+      founder.address,
+      ethers.utils.parseEther("10000000")
+    );
+    await bat["mint(address,uint256)"](
+      founder.address,
+      ethers.utils.parseEther("5001000")
+    );
+    await usdc["mint(address,uint256)"](
+      founder.address,
+      ethers.utils.parseUnits("10000000", 6)
+    );
+    await bat.transfer(cBat.address, ethers.utils.parseEther("1000000")); // We should put extra BAT to mock cBAT contract in order to provide interest
+    await bat.transfer(usdcPair.address, ethers.utils.parseEther("2000000"));
+    await usdc.transfer(
+      usdcPair.address,
+      ethers.utils.parseUnits("2000000", 6)
+    );
+    await usdcPair.mint(founder.address);
+    await addLiquidity(
+      ethers.utils.parseEther("200000"),
+      ethers.utils.parseEther("200000")
+    );
+
+    await factory.createPair(comp.address, weth.address); // Create comp and weth pair
+    const compPairAddress = factory.getPair(comp.address, weth.address);
+
+    await factory.createPair(dai.address, weth.address); // Create comp and dai pair
+    const daiPairAddress = factory.getPair(dai.address, weth.address);
+
+    compPair = new Contract(
+      compPairAddress,
+      JSON.stringify(IUniswapV2Pair.abi),
+      staker
+    ).connect(founder);
+    const daiPair = new Contract(
+      daiPairAddress,
+      JSON.stringify(IUniswapV2Pair.abi),
+      staker
+    ).connect(founder);
+
+    await dai["mint(address,uint256)"](
+      daiPair.address,
+      ethers.utils.parseEther("2000000")
+    );
+    await comp["mint(address,uint256)"](
+      compPair.address,
+      ethers.utils.parseEther("200000")
+    );
+    console.log("depositing eth to liquidity pools");
+    await weth.deposit({ value: ethers.utils.parseEther("4000") });
+    console.log(
+      await weth.balanceOf(founder.address).then((_) => _.toString())
+    );
+    await weth.transfer(compPair.address, ethers.utils.parseEther("2000"));
+    await weth.transfer(daiPair.address, ethers.utils.parseEther("2000"));
+    console.log("minting liquidity pools");
+
+    await compPair.mint(founder.address);
+    await daiPair.mint(founder.address);
```

Comment:
all of these pairs and minting could probably go to a helper, DRY DRY DRY

Reply:
Moved comppair and daipair and related